RealmSwift/Object.swift

```diff
@@ -262,6 +262,29 @@ extension Object: RealmCollectionValue {
                                    to: MutableSet<DynamicObject>.self)
     }
 
+    // MARK: Dynamic map
+
+    /**
+     Returns a map of `DynamicObject`s for a given property name.
+
+     - warning:  This method is useful only in specialized circumstances, for example, when building
+     components that integrate with Realm. If you are simply building an app on Realm, it is
+     recommended to use instance variables or cast the values returned from key-value coding.
+
+     - parameter propertyName: The name of the property.
+
+     - returns: A map of  `String` with `DynamicObject`s.
+
+     :nodoc:
+     */
+    public func dynamicMap<Key: _MapKey>(_ propertyName: String) -> Map<Key, DynamicObject> {
```

Comment:
I'm not entirely sure what the point of this method is? Could you explain what the use case would be for this to be in the public API?